src/query/api/v1/handler/prom/read.go

```diff
@@ -177,6 +251,27 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Rulemanager results are vector values (list of metric + value)
+	// Take a random number and check if under rate so we check a proportion of the queries
+	if rand.Float64() < float64(h.queryCheckConfig.CheckSampleRate) && res.Value.Type() == parser.ValueTypeVector {
```

Comment:
Can we move this check into `redis_cache.go`?  Put this a this section code into root HTTP method is exposing the storage/cache knowledge to upper layer which is not good.